src/Arr.php

```diff
@@ -726,4 +726,57 @@ public function isEmpty(): bool
     {
         return $this->length() === 0;
     }
+
+    /**
+     * Returns a string representing the elements of the array
+     * @return string the string representation
+     */
+    public function toLocaleString($locale = 'en_US.utf8', $timezone = 'UTC'): string
+    {
+        if ($this->length() <= 0) {
+            return '';
+        }
+
+        date_default_timezone_set($timezone);
+        $currentLocale = setlocale(LC_ALL, $locale);
+        $localeConfig = localeconv();
+
+        $result = $this->map(function ($item) use ($localeConfig) {
+            if (is_numeric($item)) {
+                $item = number_format(
+                    $item,
+                    is_float($item) ? 2 : 0,
+                    $localeConfig['decimal_point'] ?? '.',
+                    $localeConfig["thousands_sep"] ?? ','
+                );
+            } elseif ($this->isDate($item)) {
+                $date = date_parse($item);
+                $item = strftime(
+                    '%c',
+                    mktime($date['hour'], $date['minute'], $date['second'], $date['month'], $date['day'], $date['year'])
+                );
+            }
+
+            return $item;
+        });
+
+        return $result->toString();
+    }
+
+    /**
+     * Checks whether a value is a valid date
+     * @return boolean true if value is a valid date, false otherwise
+     */
+    protected function isDate($value): bool
```

Comment:
I personally think we should change these to private functions instead of protected methods.
Overwriting this method when inheriting it is only going to cause side effects that can be prevented by encapsulating the method.